src/librustc_mir/util/elaborate_drops.rs

```diff
@@ -703,9 +703,9 @@ impl<'l, 'b, 'tcx, D> DropCtxt<'l, 'b, 'tcx, D>
         };
 
         let cur = self.new_temp(iter_ty);
-        let length = Place::Local(self.new_temp(tcx.types.usize));
+        let length = Place::Base(PlaceBase::Local(self.new_temp(tcx.types.usize)));
         let length_or_end = if ptr_based {
-            Place::Local(self.new_temp(iter_ty))
+            Place::Base(PlaceBase::Local(self.new_temp(iter_ty)))
```

Comment:
Add a FIXME to `new_temp` to check if we want to make it return a `Place` directly if all use sites want a `Place::Base` anyway.